### PR TITLE
Parameterize generator and orchestrator codes

### PR DESCRIPTION
**Pull request by anhowe**
**What this PR does / why we need it**: Parameterize generator and orchestrator codes.

### CODE REVIEW

File: pkg/acsengine/transformtestfiles/k8s_agent_upgrade_template.json [hunk 1]
```diff
@@ -785,6 +785,22 @@
...
     },
+    "generatorCode": {
+      "defaultValue": "k8s",
+      "metadata": {
+        "description": "The generator code used to identify the generator"
+      },
+      "type": "string"
+    },
+    "orchestratorName": {
+      "defaultValue": "acsengine",
```
Comment (JackQuincy) [inline pkg/acsengine/transformtestfiles/k8s_agent_upgrade_template.json]: default values here are swapped

Comment (JackQuincy) [reply pkg/acsengine/transformtestfiles/k8s_agent_upgrade_template.json]: Same in the rest of these files.

File: pkg/armhelpers/util.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ import (
 
 const (
 	// TODO: merge with the RP code
-	k8sLinuxVMNamingFormat         = "^k8s-(.+)-([0-9a-fA-F]{8})-{0,2}([0-9]+)$"
+	k8sLinuxVMNamingFormat         = "^[0-9a-zA-Z]{3}-(.+)-([0-9a-fA-F]{8})-{0,2}([0-9]+)$"
```
Comment (JackQuincy) [inline pkg/armhelpers/util.go]: is a similar change needed for windos?

**Comment by JackQuincy** [comment]
a few small comments.

File: pkg/armhelpers/util.go (hunk 2 above)
Comment (anhowe) [reply pkg/armhelpers/util.go]: fixed

File: pkg/acsengine/transformtestfiles/k8s_agent_upgrade_template.json (hunk 1 above)
Comment (anhowe) [reply pkg/acsengine/transformtestfiles/k8s_agent_upgrade_template.json]: fixed

File: pkg/acsengine/engine.go [hunk 3]
```diff
@@ -560,6 +560,8 @@ func getParameters(cs *api.ContainerService, isClassicMode bool) (paramsMap, err
 		addValue(parametersMap, "cloudProviderRatelimitBucket", strconv.Itoa(properties.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket))
 		addValue(parametersMap, "kubeClusterCidr", properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet)
 		addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.OrchestratorProfile.KubernetesConfig.NonMasqueradeCidr)
+		addValue(parametersMap, "generatorCode", DefaultGeneratorCode)
```
Comment (amanohar) [inline pkg/acsengine/engine.go]: The service calls GenerateTemplate() and passes `api.ContainerService` (and gets back template & params) so the info what `generatorCode` and `orchestratorName` can be derived from `api.ContainerService`. That way we prevent 2 code paths how many parameters are set.

File: pkg/armhelpers/util.go [hunk 4]
```diff
@@ -68,10 +68,10 @@ func K8sLinuxVMNameParts(vmName string) (poolIdentifier, nameSuffix string, agen
 	return vmNameParts[k8sLinuxVMAgentPoolNameIndex], vmNameParts[k8sLinuxVMAgentClusterIDIndex], vmNum, nil
 }
 
-// WindowsVMNameParts returns parts of Windows VM name e.g: 50621acs9000
+// WindowsVMNameParts returns parts of Windows VM name e.g: 50621k8s9000
 func WindowsVMNameParts(vmName string) (poolPrefix string, acsStr string, poolIndex int, agentIndex int, err error) {
-	poolPrefix = strings.Split(vmName, "acs")[0]
-	poolInfo := strings.Split(vmName, "acs")[1]
+	poolPrefix = strings.Split(vmName, "k8s")[0]
```
Comment (JackQuincy) [inline pkg/armhelpers/util.go]: We can't hardcode this anymore. Either we need to accept the value to split on or we need to accept acs, k8s, or aks.  Honestly to keep existing clusters working I think we have to support handling at least acs/k8s.

Comment (anhowe) [reply pkg/armhelpers/util.go]: added a regular expression

**Pull request merged**